### PR TITLE
osd: allow osd_op_thread_timeout and suicide_timeout to be adjusted o…

### PR DESCRIPTION
**Pull request by Rethan**
…n the fly

Signed-off-by: haoyixing <haoyixing@kuaishou.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Rethan** [comment]
jenkins test make check

**Comment by Rethan** [comment]
jenkins test windows

File: src/common/WorkQueue.h [hunk 1]
```diff
@@ -598,6 +598,12 @@ class ShardedThreadPool {
     virtual void return_waiting_threads() = 0;
     virtual void stop_return_waiting_threads() = 0;
     virtual bool is_shard_empty(uint32_t thread_index) = 0;
+    void set_timeout(time_t ti) {
+      timeout_interval = ceph::make_timespan(ti);
```
Comment (ifed01) [inline src/common/WorkQueue.h:602]: Perhaps these intervals (timeout_interval/suicide_interval) worth wrapping with std::atomic to make sure they're accessed properly from multiple threads...

Comment (Rethan) [reply src/common/WorkQueue.h:602]: They could be accessed from different threads indeed. However changing the timeout configs should be infrequent, while timer was check by heartbeat frequently. So maybe it's ok to lost update once or so.

Comment (ifed01) [reply src/common/WorkQueue.h:602]: Right but IMO it's better to wrap these vars with atomic and don't care about any side effects that might occur overwise anymore...

Comment (Rethan) [reply src/common/WorkQueue.h]: WIP

**Comment by Rethan** [comment]
jenkins retest this please

**Comment by Rethan** [comment]
Has been updated to an "atomic" version.

**Review by ifed01 — APPROVED** [review]

**Comment by ljflores** [comment]
Rados suite review: https://pulpito.ceph.com/?branch=wip-yuri4-testing-2023-03-08-1234

Failures, unrelated:
    1. https://tracker.ceph.com/issues/58946
    2. https://tracker.ceph.com/issues/58560
    3. https://tracker.ceph.com/issues/58585
    4. https://tracker.ceph.com/issues/55347
    5. https://tracker.ceph.com/issues/49287

Details:
    1. cephadm: KeyError: 'osdspec_affinity' - Ceph - Orchestrator
    2. test_envlibrados_for_rocksdb.sh failed to subscribe to repo - Infrastructure
    3. rook: failed to pull kubelet image - Ceph - Orchestrator
    4. SELinux Denials during cephadm/workunits/test_cephadm - Ceph - Orchestrator
    5. podman: setting cgroup config for procHooks process caused: Unit libpod-$hash.scope not found - Ceph - Orchestrator

**Pull request merged**